sentry-core/src/main/java/io/sentry/core/SentryClient.java

```diff
@@ -65,14 +66,18 @@ public SentryClient(final @NotNull SentryOptions options, @Nullable Connection c
     random = options.getSampleRate() == null ? null : new Random();
   }
 
+  private boolean shouldApplyScopeData(final @Nullable Object hint) {
+    return (!(hint instanceof Cached) || (hint instanceof ApplyScopeData));
```

Comment:
This method being duplicated is a bit worrying since the rule of what should "applyScopeData" could change in the future and we won't realize there are two methods guarding the hint checks

Reply:
agreed, I thought to move to a static helper or something, I've written unit tests for both anyway, still, can work on it.